### ISSUE TITLE
ALSA backend virtual ports get mangled names

### ISSUE THREAD

**Issue by mzero** [mido/mido#610]
**Describe the bug**
If you open two virtual ports, one input, one output, with the rtmidi and ALSA backend, the second port gets a terrible port name.

**To Reproduce**
1. Run this program:
```py
import mido
in_port = mido.open_input("Testing", virtual=True);
out_port = mido.open_output("Testing", virtual=True);
print("ports created")
input("type anything to exit:")
```

2. Use `aconnect` or another tool to look at the names of the ports:
```sh
$ aconnect -l
...
client 129: 'RtMidiIn Client' [type=user,pid=4373]
    0 'Testing         '
client 130: 'RtMidiOut Client' [type=user,pid=4373]
    0 'RtMidiIn Client:Testing 129:0'
...
```

3. Notice that the name of the out port is **'RtMidiIn Client:Testing 129:0'**
4. Also notice that the name of the in port has been padded with spaces to 16 characters.

**Expected behavior**
Both ports should be named **'Testing'**

**Context (please complete the following information):**
- Debian 12.8
- Python version: 3.11.2
- mido package version: 1.2.10
- backend: python-rtmidi
- python-rtmidi version: 1.4.7
- librtmidi6 version: 5.0.0
- 
**Additional context**
Same issue seen on an older (debian 11) system, with mido 1.2.9, and librtmidi5 4.0.0

**Comment by rdoursenaud** [comment]
Thanks for the detailed report.

Unfortunately I don’t have a Linux machine handy to try and reproduce immediately.

From looking at your sample and around the code, it seems that the first opened port somewhat polutes the name of the second (Padding nonwithstanding as I think it’s an upstream issue).

Could you please help me test this hypothesis by posting the results when creating the ports in the reverse order (output before input): 
```py
import mido
out_port = mido.open_output("Testing", virtual=True)
in_port = mido.open_input("Testing", virtual=True)
```

Also using an `ioport` may yield better results and could be a good workaround for you if it works:
```py
io_port = mido.open_ioport("Testing", virtual=True)
# Individual ports are available as `io_port.input` and `io_port.output`.
```

Finally, could you please test using the latest released mido version 1.3.2?

Looking forward to reading from you.

**Comment by mzero** [comment]
The reversing the order just changes which port gets the bad name:

```
client 129: 'RtMidiOut Client' [type=user,pid=294916]
    0 'Testing         '
client 131: 'RtMidiIn Client' [type=user,pid=294916]
    0 'RtMidiOut Client:Testing 129:0'
```

I didn't test against 1.3.2... but I found the bug... and then found that it has been fixed in main:

Turns out this is a duplicate of this bug https://github.com/mido/mido/issues/471
And that it was fixed with this commit: https://github.com/mido/mido/commit/70b9a1718d32c93410ce8a19bca4668d4003186e

And indeed, that fix is in 1.3.2

Sadly, 1.2.10 is what is in current Debian... and planned for next debian.

I agree that `open_ioport` would be a better choice... but I found this bug because another piece of software did it with two ports (touchosc2midi, which reported the original bug).

Anyway - sorry I didn't see that this was a duplicate before

**Comment by rdoursenaud** [comment]
Thank you for your feedback and sorry it caused you trouble.

I had a hunch it was something I already worked on.

Unfortunately, distribution packaging is out of our hands and Debian is known to be *very* conservative when it comes to new versions. It can take years for versions to trickle down into it.
It’s both a blessing and a curse.
I’ve been an avid user on servers for years and it’s great for that use case.
For making music however, I’d choose something else.
Sorry, I digress…

Debian patches most of their packages heavily already.
Maybe ask the Debian maintainer of the package to backport the fix?

I’ll close the bug for now but tet us know if I can be of any help.

Duplicate #471
